src/test/java/ro/developmentfactory/thedrones/service/DronePositionServiceImplTest.java

```diff
@@ -0,0 +1,178 @@
+package ro.developmentfactory.thedrones.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import ro.developmentfactory.thedrones.controller.dto.DroneStatusResponse;
+import ro.developmentfactory.thedrones.repository.DroneRepository;
+import ro.developmentfactory.thedrones.repository.DroneStatusRepository;
+import ro.developmentfactory.thedrones.repository.entity.Direction;
+import ro.developmentfactory.thedrones.repository.entity.Drone;
+import ro.developmentfactory.thedrones.repository.entity.DroneStatus;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class DronePositionServiceImplTest {
```

Comment:
the tests are good, I would probably only recommend adding more cases to capture all edges of the field to make sure they all are handled. it's quite easy if you rewrite to parametrized tests - separately valid move forward and invalid. you could also quite easily cover all with parametrized tests for turning, both directions. such tests are easy to write and they usually save the code a lot in production

Reply:
please add back these tests, you need them